File: unique_char.swift
// find unique characters in a string

func uniqueChar(string:String) -> Int {
  let stringChars = Array(string)
  var uniqChars:[String] = []
  for var i = 0; i < stringChars.count; i++ {
    if !contains(uniqChars,String(stringChars[i])) {
      uniqChars.append(String(stringChars[i]))
    }
  }
  return uniqChars.count
}

println(uniqueChar("abracadabra"))

File: unique_char.rb
# find unique characters in a string in ruby

def unique_chars(string)
  unique_characters = []
  (0...string.length).each { |i| unique_characters << string[i] if !unique_characters.include?(string[i]) }
  unique_characters.count
end

p unique_chars("abracadabra")
